tensorboard/plugins/debugger_v2/debug_data_multiplexer.py

```diff
@@ -446,6 +451,104 @@ def GraphExecutionData(self, run, begin, end, trace_id=None):
             ],
         }
 
+    def GraphOpInfo(self, run, graph_id, op_name):
+        """Get the information regarding a graph op's creation.
+
+        Args:
+          run: Name of the run.
+          graph_id: Debugger-generated ID of the graph that contains
+            the op in question. This ID is available from other methods
+            of this class, e.g., the return value of `GraphExecutionDigests()`.
+          op_name: Name of the op.
+
+        Returns:
+          A JSON-serializable object containing the information regarding
+          the op's creation.
+
+        Raises:
+          NotFoundError if the graph_id or op_name does not exist.
+        """
+        runs = self.Runs()
+        if run not in runs:
+            return None
+        try:
+            graph = self._reader.graph_by_id(graph_id)
```

Comment:
From this perspective, this call looks like it could be expensive, i.e. you wouldn't want to reload the graph for every per-op request.  In practice I see it's backed by a dict (i.e. the graphs are already in memory) so it's OK, but I wonder if the API could be clearer about this somehow.  Maybe the issue is that the call is on `self._reader`, so I expect it to *read* something.  Should that be called `self._debug_data` or something?  Or, should `graph_by_id` really by `graph_reader_by_id`, to suggest (IIUC) that what you get is basically a sub-reader?  (I don't know a good answer, just speculating)

Reply:
Thanks for the comment. As you pointed out `graph_by_id()` is cheap for the current local-only case, because it's just a `dict` lookup. In a future hosted setting, this will be more complex. It'll most likely involve a database (Spanner query) with graph_id and op_name as a composite key or a single key, but the details of that is yet to be worked out.

The name `_reader` is appropriate in the sense that the object does read data from the tfdbg2-format files. There is some confusion because it does not only read the data, but organizes the data (like building an internal index for consumers of ops). So perhaps the name can be improved to reflect that, but I don't know what a good alternative name should be. In any case, the naming issue is pre-existing and we can keep discussing it without blocking this PR IMO.

Reply:
ok

Reply:
Ack.